test/unit-test/cellular_at_core_utest.c

```diff
@@ -0,0 +1,932 @@
+    cellularStatus = Cellular_ATIsStrDigit( NULL, pResult );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+
+    /* Check that CELLULAR_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATIsStrDigit( pString, NULL );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+}
+
+/**
+ * @brief Test happy path for Cellular_ATIsStrDigit to return CELLULAR_AT_SUCCESS and got expected results.
+ */
+void test_Cellular_ATIsStrDigit_Happy_Path( void )
+{
+    CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
+    char pString[] = CELLULAR_SAMPLE_DIGIT_STRING;
+    char pStringWithNull[] = CELLULAR_SAMPLE_DIGIT_STRING_WITH_NULL;
+    bool Result;
+
+    /* Check that CELLULAR_AT_SUCCESS is returned. */
+    cellularStatus = Cellular_ATIsStrDigit( pString, &Result );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_SUCCESS, cellularStatus );
+    TEST_ASSERT_EQUAL( true, Result );    
+
+    /* Check that CELLULAR_AT_SUCCESS is returned. */
+    cellularStatus = Cellular_ATIsStrDigit( pStringWithNull, &Result );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_SUCCESS, cellularStatus );
+}
+
+/**
+ * @brief Test the error path for Cellular_ATIsStrDigit to return CELLULAR_AT_BAD_PARAMETER.
+ */
+void test_Cellular_ATIsStrDigit_Error_Path( void )
+{
+    CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
+    char * pString = "";
+    bool Result;
+
+    /* Check that CELLULAR_AT_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATIsStrDigit( pString, &Result );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+}
+
+/**
+ * @brief Test that any NULL parameter causes Cellular_ATcheckErrorCode to return CELLULAR_AT_BAD_PARAMETER.
+ */
+void test_Cellular_ATcheckErrorCode_Invalid_Param( void )
+{
+    CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
+    char * pString;
+    bool * pResult;
+    const char * const * const ppKeyList = {"TEST1:SUCCESS", "TEST2:ERROR"};
+    size_t keyListLen = 2;
+
+    /* Check that CELLULAR_AT_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATcheckErrorCode( NULL, NULL, keyListLen, NULL );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+
+    /* Check that CELLULAR_AT_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATcheckErrorCode( pString, NULL, keyListLen, pResult );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+}
+
+/**
+ * @brief Test happy path for Cellular_ATcheckErrorCode to return CELLULAR_AT_SUCCESS and got expected results.
+ */
+void test_Cellular_ATcheckErrorCode_Happy_Path( void )
+{
+    CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
+    char pString[] = CELLULAR_SAMPLE_SUCCESS_CODE_STRING;
+    char pStringError[] = CELLULAR_SAMPLE_ERROR_CODE_STRING;
+    char * ppKeyList[] = {"TEST1:SUCCESS", "TEST2:ERROR"};
+    size_t keyListLen = 2;
+    bool Result;
+
+    /* Check that CELLULAR_AT_BAD_PARAMETER is returned. */

```

Comment:
Comment and code are not in sync. In fact, these comments do not add any value as the code is self-explanatory. We can remove all these `Check that xxx is returned` comments.

Reply:
Sure.